shell/platform/linux/public/flutter_linux/fl_binary_messenger.h

```diff
@@ -0,0 +1,126 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_BINARY_MESSENGER_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_BINARY_MESSENGER_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <gio/gio.h>
+#include <glib-object.h>
+
+G_BEGIN_DECLS
+
+G_DECLARE_FINAL_TYPE(FlBinaryMessenger,
+                     fl_binary_messenger,
+                     FL,
+                     BINARY_MESSENGER,
+                     GObject)
+
+/**
+ * FlBinaryMessenger:
+ *
+ * #FlBinaryMessenger is an object that allows sending and receiving of platform
+ * messages with a #FlEngine.
+ */
+
+/**
+ * FlBinaryMessengerResponseHandle:
+ *
+ * A handle used to respond to platform messages.
+ */
+typedef struct _FlBinaryMessengerResponseHandle FlBinaryMessengerResponseHandle;
+
+/**
+ * FlBinaryMessengerCallback:
+ * @messenger: a #FlBinaryMessenger
+ * @channel: channel message received on
+ * @message: message content received from Dart
+ * @response_handle: a handle to respond to the message with
+ * @user_data: (closure): data provided when registering this callback
+ *
+ * Function called when platform messages are received.
```

Comment:
This comment should also indicate that replying is mandatory; the more relevant places it's mentioned, the better.